nss/integration-tests/nss_test.go

```diff
@@ -0,0 +1,176 @@
+		"Error when getting passwd by name and daemon is not available": {db: "passwd", key: "user1", noDaemon: true, wantErr: true},
+		"Error when getting group by name and daemon is not available":  {db: "group", key: "group1", noDaemon: true, wantErr: true},
+		"Error when getting shadow by name and daemon is not available": {db: "shadow", key: "user1", noDaemon: true, wantErr: true},
+
+		"Error when getting passwd by id and daemon is not available": {db: "passwd", key: "1111", noDaemon: true, wantErr: true},
+		"Error when getting group by id and daemon is not available":  {db: "group", key: "11111", noDaemon: true, wantErr: true},
+
+		"Error when getting passwd by name and entry does not exist": {db: "passwd", key: "doesnotexit", wantErr: true},
+		"Error when getting group by name and entry does not exist":  {db: "group", key: "doesnotexit", wantErr: true},
+		"Error when getting shadow by name and entry does not exist": {db: "shadow", key: "doesnotexit", wantErr: true},
+
+		"Error when getting passwd by id and entry does not exist": {db: "passwd", key: "404", wantErr: true},
+		"Error when getting group by id and entry does not exist":  {db: "group", key: "404", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			if tc.cacheDB == "" {
+				tc.cacheDB = "multiple_users_and_groups"
+			}
+
+			var socketPath string
+			if !tc.noDaemon && !tc.noCustomSocket {
+				ctx, cancel := context.WithCancel(context.Background())
+				socketPath = runDaemon(ctx, t, tc.cacheDB)
+				t.Cleanup(cancel)
+			}
+
+			cmds := []string{"getent", tc.db}
+			if tc.key != "" {
+				cmds = append(cmds, tc.key)
+			}
+
+			got, err := outNSSCommandForLib(t, socketPath, originOuts[tc.db], cmds...)
+
+			// Some tests use corrupted databases to trigger specific errors. When opening a corrupted database,
+			// the cache will schedule a clear-up routine to fix the database, so we need to give it some time to
+			// finish before proceeding with the test, otherwise the test cleanup functions would run (removing
+			// the files and directories) before the routine has finished and it will panic.
+			if tc.cacheDB != "multiple_users_and_groups" {
+				time.Sleep(time.Second)
+			}
+
+			if tc.wantErr {
+				require.Error(t, err, "Expected an error, but got none")
+				return
+			}
+			require.NoError(t, err, "Expected no error, but got %v", err)
+
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "Outputs must match")
+		})
+	}
+}
+
+func TestMain(m *testing.M) {
+	testutils.InstallUpdateFlag()
+	flag.Parse()
+
+	execPath, cleanup, err := buildDaemon()
+	if err != nil {
+		log.Printf("Setup: failed to build daemon: %v", err)
+		os.Exit(1)
+	}
+	defer cleanup()
+	daemonPath = execPath
+
+	// Creates the directory to store the coverage files for the integration tests.
+	if testutils.WantCoverage() {
+		rawCovDir = os.Getenv("GOCOVERDIR")
```

Comment:
I am surprise by the use of the env variable here (in the parent process). We will set it in the child one for sure, but we shouldn’t require it in the parent one.

Or is it the use of -cover in go tests which sets the env variable internally?

Reply:
We are not requiring the env in the parent process. This is mainly to check if the var is set and then, if it is, we'll use its value (which is a path) to store the coverage files (a bit similar to what we do with TEST_RUST_TARGET, even though this one is to define the build target for Rust), if it's not set we create a new directory and use it instead.

Reply:
ack!